modules/skybrowser/include/wwtcommunicator.h

```diff
@@ -0,0 +1,112 @@
+ ****************************************************************************************/
+
+#ifndef __OPENSPACE_MODULE_SKYBROWSER___WWTCOMMUNICATOR___H__
+#define __OPENSPACE_MODULE_SKYBROWSER___WWTCOMMUNICATOR___H__
+
+#include <modules/skybrowser/include/browser.h>
+
+#include <openspace/properties/vector/ivec3property.h>
+#include <openspace/properties/vector/dvec2property.h>
+#include <openspace/properties/scalar/floatproperty.h>
+#include <openspace/documentation/documentation.h>
+#include <deque>
+
+namespace openspace {
+
+class WwtCommunicator : public Browser {
+public:
+    explicit WwtCommunicator(const ghoul::Dictionary& dictionary);
+    WwtCommunicator(const WwtCommunicator&) = default;
+    ~WwtCommunicator();
+
+    void update();
+    void render();
+    void initializeGL();
+    void deinitializeGL();
+
+    // WorldWide Telescope communication
+    void displayImage(const std::string& url, int i);
+    void removeSelectedImage(int i);
+    void setImageOrder(int i, int order);
+    void loadImageCollection(const std::string& collection);
+    void setImageOpacity(int i, float opacity);
+    void hideChromeInterface(bool shouldHide);
+
+    bool hasLoadedImages() const;
+    float verticalFov() const;
+    glm::ivec3 borderColor() const;
+    glm::dvec2 equatorialAim() const;
+    glm::dvec2 fieldsOfView() const;
+    const std::deque<int>& getSelectedImages() const;
+
+    void setHasLoadedImages(bool isLoaded);
+    void setVerticalFov(float vfov);
+    void setIsSyncedWithWwt(bool isSynced);
+    void setEquatorialAim(glm::dvec2 equatorial);
+    void setBorderColor(glm::ivec3 color);
+
+    void highlight(glm::ivec3 addition);
+    // The removal parameter decides what will be removed from the border color
+    void removeHighlight(glm::ivec3 removal);
+    void updateBorderColor();
+    void updateAim();
+    
+protected:
+    void setIdInBrowser(const std::string& id);
+
+    float _verticalFov = 10.f;
+    glm::ivec3 _borderColor = glm::ivec3(70);
+    glm::dvec2 _equatorialAim = glm::dvec2(0.0);
+    bool _hasLoadedImages = false;
+    std::deque<int> _selectedImages;
+
+private:
+    void setWebpageBorderColor(glm::ivec3 color);
+    void sendMessageToWwt(const ghoul::Dictionary& msg);
+
+    // WorldWide Telescope messages
+    ghoul::Dictionary moveCameraMessage(const glm::dvec2& celestCoords, double fov,
+        double roll, bool shouldMoveInstantly = true);
+    ghoul::Dictionary loadCollectionMessage(const std::string& url);
+    ghoul::Dictionary setForegroundMessage(const std::string& name);
+    ghoul::Dictionary addImageMessage(const std::string& id, const std::string& url);
+    ghoul::Dictionary removeImageMessage(const std::string& id);
+    ghoul::Dictionary setImageOpacityMessage(const std::string& id, double opacity);
+    ghoul::Dictionary setLayerOrderMessage(const std::string& id, int version);
+    ghoul::Dictionary hideChromeGuiMessage(bool isHidden); // Requires a newer CEF version 
```

Comment:
Is there a plan to update the CEF version then? A bit confused, as this function seems to be used, but is it not working? 🤔

Reply:
Yep Micah had already updated CEF-browser but it is not finished. Figured I could leave the function there in that case